sktime/forecasting/dynamic_factor.py

```diff
@@ -0,0 +1,429 @@
+        using finite difference methods use a centered approximation. Default is False.
+    method : str , 'lbfgs'
+        ‘newton’ for Newton-Raphson
+        ‘nm’ for Nelder-Mead
+        ‘bfgs’ for Broyden-Fletcher-Goldfarb-Shanno (BFGS)
+        ‘lbfgs’ for limited-memory BFGS with optional box constraints
+        ‘powell’ for modified Powell’s method
+        ‘cg’ for conjugate gradient
+        ‘ncg’ for Newton-conjugate gradient
+        ‘basinhopping’ for global basin-hopping solver
+    maxiter : int , optional ,default = 50
+        The maximum number of iterations to perform.
+    full_output : bool , default = 1
+        Set to True to have all available output in the
+        Results object’s mle_retvals attribute.
+        The output is dependent on the solver.
+    disp : bool ,   default = 5
+        Set to True to print convergence messages.
+    callback : callable callback(xk) , default = None
+        Called after each iteration, as callback(xk),
+        where xk is the current parameter vector.
+    return_params : bool ,default = False
+        Whether or not to return only the array of maximizing parameters.
+    optim_score : {'harvey','approx'} , default = None
+        The method by which the score vector is calculated. ‘harvey’ uses the method
+        from Harvey (1989), ‘approx’ uses either finite difference or
+        complex step differentiation depending upon the value of optim_complex_step,
+        and None uses the built-in gradient approximation of the optimizer.
+    optim_complex_step : bool , default = True
+        Whether or not to use complex step differentiation
+        when approximating the score; if False, finite difference approximation is used.
+    optim_hessian : {'opg','oim','approx'} , default = None
+        ‘opg’ uses outer product of gradients, ‘oim’ uses the information
+        matrix formula from Harvey (1989), and ‘approx’ uses numerical approximation.
+    low_memory : bool , default = False
+        If set to True, techniques are applied to substantially reduce memory usage.
+        If used, some features of the results object will not be available
+        (including smoothed results and in-sample prediction),
+        although out-of-sample forecasting is possible.
+
+    See Also
+    --------
+    statsmodels.tsa.statespace.dynamic_factor.DynamicFactor
+    statsmodels.tsa.statespace.dynamic_factor.DynamicFactorResults
+    References
+    ----------
+    [1] Lütkepohl, Helmut. 2007. New Introduction to Multiple Time Series Analysis.
+    Berlin: Springer.
+    """
+
+    _tags = {
+        "scitype:y": "multivariate",
+        "ignores-exogeneous-X": False,
+        "handles-missing-data": True,
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "capability:pred_int": False,

```

Comment:
hm, I thought, based on the formulation, that the model should be able to make proba predictions?

Though, I cannot find a method for that, so maybe let's leave this to another PR but make a note in the issue?